
on_failure.reason_invalid_auth_responses = "Auth is not allowed"
on_failure.sip_code_invalid_auth_responses = 488

on_failure.reason_branch_timeout = "Request Timeout"
on_failure.sip_code_branch_timeout = 408

failure_route[MANAGE_FAILURE] {

    if (t_is_canceled() || t_is_expired()) drop();

    route(HANDLE_NAT_REPLY);

    if(t_check_status("401|407")) {
        ##xlog("L_WARN", "[MANAGE_FAILURE]: Auth response received: $T_rpl($rs) \n");
        send_reply("$ccp(on_failure.sip_code_invalid_auth_responses)", "$ccp(on_failure.reason_invalid_auth_responses)");
        exit;
    }

    if (t_branch_timeout() && !t_branch_replied()) {
        xlog("L_WARN", "[MANAGE_FAILURE]: TIMEOUT request $rm $ru, from $fu, [$(mb{s.len}) bytes] ($ci) \n");
        send_reply("$ccp(on_failure.sip_code_branch_timeout)", "$ccp(on_failure.reason_branch_timeout)");
        exit;
    }

}
